employees/views.py

```diff
@@ -201,7 +201,7 @@ def _handle_join_request(self, request: HttpRequest) -> HttpResponse:
             logger.debug(f"User with id: {request.user.pk} join to the project with id: {project.pk}")
             return redirect(self.get_success_url())
         else:
-            return self.render_to_response(context={"form": form})
+            return self.render_to_response(context=self.get_context_data())
```

Comment:
I think that now you are not returning invalid project form and user will not get any information what happened. I am not sure what solution would be better but maybe you could `update` this dictionary ?

Reply:
It's a good point. Truth be told, I don't know how this would affect the view though. This was created in case an empty form is submitted. Normally this is impossible, since empty form is created when there are no projects to join to. But in such case, the form is hidden from user in the view, so form's error feedback is useless. Although in case of a different validation error (which I can't think of right now, but since we're considering manual requests here I guess more are possible) it could be different.

I suppose at the very least for formality's sake, the validated form should be passed to response.
@rwrzesien @kbeker Would that be the most appropriate approach?

Reply:
I see this is resolved already?